pkg/operator/controller/status/controller.go

```diff
@@ -685,3 +729,45 @@ func operatorStatusesEqual(a, b configv1.ClusterOperatorStatus) bool {
 
 	return true
 }
+
+// joinConditions merges two cluster operator conditions into one.
+// If both conditions have the same status:
+// - Reason becomes: "MultipleComponents[Not]" + condType.
+// - Messages are concatenated.
+// If the new condition has a higher-priority status, it overrides the current one.
+// Status priority (highest to lowest): True, Unknown, False, empty.
+func joinConditions(currCond, newCond configv1.ClusterOperatorStatusCondition, condType string) configv1.ClusterOperatorStatusCondition {
+	switch newCond.Status {
+	case configv1.ConditionTrue:
+		if currCond.Status == configv1.ConditionTrue {
+			currCond.Reason += "And" + newCond.Reason
+			currCond.Message += newCond.Message
```

Comment:
```suggestion
			currCond.Reason += " And " + newCond.Reason
			currCond.Message += ", " + newCond.Message
```

Reply:
I fixed this a little differently in my PR https://github.com/openshift/cluster-dns-operator/pull/373/files#diff-81f08fdfab85d1173ed67b51c2d429a60efbfb17e3dc6ef01ed0afe2011023f2R488

I fixed it because it looked like gibberish in the CI events graph.

Reply:
Ack. I still prefer to stay with `Reason` message to not have white spaces because I believe this is Kubernetes native to use Camel case concatenation in cases of long messages (ex: `PodReadyToStartContainers`, `CrashLoopBackOff` etc.).

Reply:
I prefer the way we did it for computing the Progressing condition: https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/status/controller.go#L574-L576 because it is easier to read and review.